frontend/src/components/User/Login.jsx

```diff
@@ -46,12 +53,36 @@ export default function LoginPage() {
       .then(() => setStatus("complete"));
   };
 
+  const doGithubLogin = (code) => {
+    setStatus("loading");
+    sendRequest("/github", { code }).then(
+      (user) => {
+        setUserProfile(user);
+        setStatus("githubAuth");
+      },
+      (err) => {
+        setComplete({
+          rstatus: "error",
+          msg: err.message,
+          buttonFn: () => {
+            setStatus("githubAuth");
```

Comment:
If user fail to login with github and the button redirect to `login` state, it will send the same request again, such as `/login?code=xxx`. And this will fail as well. Now the button redirect to `GithubLoginFinish` state and without user logged in. This will lead to a page jump to the login page without `codeID`. I don't think it's a graceful implementation and I will refine it later.

Reply:
I see. The reason it could work is that the dashboard page [will check the user login status](https://github.com/openyurtio/yurt-dashboard/blob/73f4e03b458a769950aedcf21b240286ec9ea30b/frontend/src/components/Utils/ContentWrapper.jsx?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L134) and redirect to login page. It will go to `login` page finally as expected with two hops. The key is to rewrite the `location` to remove the `code=xxx` part. We can ignore it for now and consider optimizing it in the next release.